### PR TITLE
Upgrade angular project version to 8.0

### PR DESCRIPTION
**Pull request by SergiiVlasiuk**
Upgrade angular project version to 8.0:
-
- fix build;
- fix dependencies;
- fix `jasmine` tests;
- fix `protractor` lounch;

TODO:
-
- `fix protractor test`;
- left default readme notes, because of no permissions and unclear some in original part.

### CODE REVIEW

File: e2e/protractor.conf.js [hunk 1]
```diff
@@ -1,14 +1,16 @@
...
+const { SpecReporter } = require('jasmine-spec-reporter');
 
+/**
+ * @type { import("protractor").Config }
+ */
 exports.config = {
   allScriptsTimeout: 11000,
   specs: [
-    './e2e/**/*.e2e-spec.ts'
+    '../e2e/**/*.e2e-spec.ts'
```
Comment (lydemann) [inline e2e/protractor.conf.js]: Maybe we could just do:
```suggestion
    './**/*.e2e-spec.ts'
```

File: e2e/protractor.conf.js [hunk 2]
```diff
@@ -19,16 +21,12 @@ exports.config = {
   jasmineNodeOpts: {
     showColors: true,
     defaultTimeoutInterval: 30000,
-    print: function () {}
+    print: function () { }
   },
   onPrepare() {
     require('ts-node').register({
-      project: require('path').join(__dirname, './tsconfig.e2e.json')
+      project: require('path').join(__dirname, './tsconfig.json')
```
Comment (lydemann) [inline e2e/protractor.conf.js]: Why change name here?

File: e2e/tsconfig.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
   "extends": "../tsconfig.json",
   "compilerOptions": {
-    "outDir": "../out-tsc/app",
+    "outDir": "../out-tsc/e2e",
```
Comment (lydemann) [inline e2e/tsconfig.json]: Why change path name here?

File: e2e/protractor.conf.js (hunk 2 above)
Comment (SergiiVlasiuk) [reply e2e/protractor.conf.js]: default name generated by `ng-cli`

File: e2e/tsconfig.json (hunk 3 above)
Comment (SergiiVlasiuk) [reply e2e/tsconfig.json]: default name generated by `ng-cli`

File: README.md [hunk 4]
```diff
@@ -1,33 +1,11 @@
-# AngularBestPracticesDemo
-A demo of Angular best practices including:
+# AngularDemoWithBestPracticesV8
 
-- VSCode config for fix on saving and organize imports
```
[the remark this answers is not in the document]
Comment (SergiiVlasiuk) [reply README.md]: Sure, but possible it requires different details or more explanation to make it more clear. (i'm sorry for my bad understanding...)

...any way i'll fix your recommendations if i have some time for that.

**Comment by lydemann** [comment]
@SergiiVlasiuk I have updated the tslint rules to my recommended tslint rules. Thanks again for the help here!

**Pull request merged**